amgut/lib/data_access/ag_data_access.py

```diff
@@ -203,12 +168,13 @@ def getAGBarcodeDetails(self, barcode):
                   INNER JOIN ag_login USING (ag_login_id)
                   INNER JOIN barcode USING (barcode)
                   WHERE barcode = %s"""
-        row = self._sql.execute_fetchone(sql, [barcode])
 
-        row_dict = {}
-        if row:
-            row_dict = dict(row)
-        return row_dict
+        with TRN:
+            TRN.add(sql, [barcode])
+            row = TRN.execute_fetchindex()
+            if not row:
+                raise ValueError('Barcode does not exist in AG: %s' % barcode)
+            return dict(row[0])
 
```

Comment:
The signature of this method has changed. Previously, an empty `dict` was possible whereas now it raises. Is that expected?